server/client_server.go

```diff
@@ -22,21 +23,126 @@ func newServerClient(jiraClient *jira.Client) Client {
 	}
 }
 
+type IssueInfo struct {
+	Values []*jira.MetaIssueType `json:"values,omitempty"`
+}
+
+type FieldInfo struct {
+	Values []interface{} `json:"values,omitempty"`
+}
+
+type FieldValues struct {
+	Id string `json:"fieldId,omitempty"`
+}
+
+type FieldId struct {
+	Values []FieldValues `json:"values,omitempty"`
+}
+type Version struct {
+	VersionInfo string `json:"version,omitempty"`
+}
+
 // GetCreateMeta returns the metadata needed to implement the UI and validation of
 // creating new Jira issues.
 func (client jiraServerClient) GetCreateMeta(options *jira.GetQueryOptions) (*jira.CreateMetaInfo, error) {
-	cimd, resp, err := client.Jira.Issue.GetCreateMetaWithOptions(options)
+	apiEndpoint := "rest/api/2/serverInfo"
+	v := new(Version)
+	req, err := client.Jira.NewRequest("GET", apiEndpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+	_, err = client.Jira.Do(req, v)
+	if err != nil {
+		return nil, err
+	}
+
+	v1, err := version.NewVersion(v.VersionInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	v2, err := version.NewVersion("8.4.0")
+	if err != nil {
+		return nil, err
+	}
+
+	var info *jira.CreateMetaInfo
+	var response *jira.Response
+	if v1.LessThan(v2) {
+		cimd, resp, error := client.Jira.Issue.GetCreateMetaWithOptions(options)
+		info = cimd
+		response = resp
+		err = error
+	} else {
+		cd, resp, err := client.Jira.Project.ListWithOptions(options)
+		meta := new(jira.CreateMetaInfo)
+
+		if err == nil {
+			for i := 0; i < len(*cd); i++ {
+				meta.Expand = (*cd)[i].Expand
+				project := new(jira.MetaProject)
+				project.Self = (*cd)[i].Self
+				project.Id = (*cd)[i].ID
+				project.Key = (*cd)[i].Key
+				project.Name = (*cd)[i].Name
+				apiEndpoint := "rest/api/2/issue/createmeta/" + (*cd)[i].ID + "/issuetypes"
+				req, err := client.Jira.NewRequest("GET", apiEndpoint, nil)
+				if err != nil {
+					break
+				}
+
+				issues := new(IssueInfo)
+				resp, err = client.Jira.Do(req, issues)
+				if err != nil {
+					break
+				}
+
+				project.IssueTypes = issues.Values
+				for j := 0; j < len(project.IssueTypes); j++ {
+					apiEndpoint := "rest/api/2/issue/createmeta/" + (*cd)[i].ID + "/issuetypes/" + project.IssueTypes[j].Id
+					req, err := client.Jira.NewRequest("GET", apiEndpoint, nil)
+					if err != nil {
+						break
+					}
+
+					field := new(FieldInfo)
+					resp, err = client.Jira.Do(req, field)
+					if err != nil {
+						break
+					}
+
+					fieldId := new(FieldId)
```

Comment:
In go for abbreviations with use all the alphabets as capital keep this in mind and change this everywhere
```suggestion
					fieldID := new(FieldId)
```